### DISCUSSION TITLE
Exporting type annotations to lua?

### DISCUSSION THREAD

**Discussion by eigenbom** [teal-language/tl#613] [General]
Has there been any thought or discussion around exporting type spec/annotations from teal to lua? (or offering hooks to users to do this?) E.g., 

```lua
-- add.tl
local function add(a: number, b: number): number
   return a + b
end

-- might export to add.lua...
---@param a number
---@param b number
local function add(a, b)
   return a + b
end
```

**Comment by hishamhm** [comment]
@eigenbom no plans for that on my end, though I would like to see Teal-specific javadoc/ldoc-style documentation tools emerging.

Generally, I don't consider it a goal for the generated .lua to be particularly readable or nice-looking. It currently happens to be, mostly for ease of my own debugging, and I don't plan to go out of my way to make the generated code _less_ readable. 

However, if you look at other languages that do source-to-source compilation such as TypeScript, the more complex the transformations the compiler needs to do, the more the code starts looking "machine-generated". I don't want to close the door on the possibility of more complex transformations in the future, so for now I also I don't intend to add features that add to the generated code.

**Comment by eigenbom** [comment]
Thx for clarifying. I don't mind if the generated code is complex, but it would be nice to output some kind of nice doc/meta file (e.g. https://github.com/sumneko/lua-language-server/blob/master/meta/3rd/love2d/library/love.lua) so people can export documented lua libraries.
